decidim-core/app/models/decidim/organization.rb

```diff
@@ -22,12 +22,12 @@ class Organization < ApplicationRecord
 
     has_many :static_pages, foreign_key: "decidim_organization_id", class_name: "Decidim::StaticPage", inverse_of: :organization, dependent: :destroy
     has_many :static_page_topics, class_name: "Decidim::StaticPageTopic", inverse_of: :organization, dependent: :destroy
-    has_many :scopes, -> { order(name: :asc) }, foreign_key: "decidim_organization_id", class_name: "Decidim::Scope", inverse_of: :organization
-    has_many :scope_types, -> { order(name: :asc) }, foreign_key: "decidim_organization_id", class_name: "Decidim::ScopeType", inverse_of: :organization
-    has_many :areas, -> { order(name: :asc) }, foreign_key: "decidim_organization_id", class_name: "Decidim::Area", inverse_of: :organization
-    has_many :area_types, -> { order(name: :asc) }, foreign_key: "decidim_organization_id", class_name: "Decidim::AreaType", inverse_of: :organization
-    has_many :admins, -> { where(admin: true) }, foreign_key: "decidim_organization_id", class_name: "Decidim::User"
-    has_many :users_with_any_role, -> { where.not(roles: []) }, foreign_key: "decidim_organization_id", class_name: "Decidim::User"
+    has_many :scopes, -> { order(name: :asc) }, foreign_key: "decidim_organization_id", class_name: "Decidim::Scope", inverse_of: :organization, dependent: :destroy
```

Comment:
I think this is fine as when the Organization is destroyed, all related records are destroyed but this won't apply to everything.

I believe that e.g. proposals are left behind as we don't define `dependent: :destroy` here:
https://github.com/decidim/decidim/blob/7b2a3d6f48175c42a3a0fc7a821d8ff876c07aea/decidim-core/lib/decidim/has_component.rb#L12

And for the same reason I believe participatory processes (and other spaces?) are also left behind:
https://github.com/decidim/decidim/blob/7b2a3d6f48175c42a3a0fc7a821d8ff876c07aea/decidim-participatory_processes/app/models/decidim/participatory_process.rb#L30-L32

That said, I don't think it's a big problem as I wouldn't expect deleting an organization to happen very often. I have never even tried what happens when we delete an organization and I wouldn't be surprised if there are some other problems with that too.

Reply:
Well, I only added Dependent destroy only where rubocop complained about it ...